### PR TITLE
Fix the compiler lib builds for multiconfig generators

### PR DESCRIPTION
**Pull request by sstamenova**
When the code was separated into its own library, the generator expression for the file configuration was dropped. This is needed because file generation happens at cmake config time and for multiconfig generators such as VS or Xcode the build configuration is not known.

This change brings back the generator expression for file generation.

Signed-off-by: Stella Stamenova <stilis@microsoft.com>

### CODE REVIEW

File: src/Compiler/CMakeLists.txt [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+# CMAKE_CFG_INTDIR is . for single-config generators such as make, and
+# it has a value (e.g. $(Configuration)) otherwise, so we can use it to
+# determine whether we are dealing with a multi-config generator.
+if (NOT "${CMAKE_CFG_INTDIR}" STREQUAL ".")
+  set(FILE_GENERATE_DIR ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR})
+else()
+  set(FILE_GENERATE_DIR ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_BUILD_TYPE})
+endif()
+
 add_custom_target(ExternalUtil DEPENDS ${FILE_GENERATE_DIR}/ExternalUtil.hpp)
```
Comment (gongsu832) [inline src/Compiler/CMakeLists.txt:24]: This should also be changed to `${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/ExternalUtil.hpp`?

Comment (sstamenova) [reply src/Compiler/CMakeLists.txt:24]: No, the targets get evaluated at build time when the configuration is known, so it shouldn't use a generator expression. The `file` call gets evaluated at cmake configuration time, so it needs to use the generator expression.

Comment (gongsu832) [reply src/Compiler/CMakeLists.txt:24]: Oh right. You set `FILE_GENERATOR_DIR` above. Sorry, for some reason I can never read the github diff on the web page properly.

**Comment by AlexandreEichenberger** [comment]
@gongsu832 will defer to you on accepting this patch

**Review by gongsu832 — APPROVED** [review]
LGTM

**Pull request merged**